### PR TITLE
restrict more uri variables

### PR DESCRIPTION
**Pull request by sebastiankb**
also see #1068

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/pull/1109.html" title="Last updated on Apr 28, 2021, 9:46 AM UTC (767f5fc)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/wot-thing-description/1109/24c2bf8...767f5fc.html" title="Last updated on Apr 28, 2021, 9:46 AM UTC (767f5fc)">Diff</a>

### CODE REVIEW

**Comment by egekorkan** [comment]
Thanks for fixing the weird numbers that were popping up as well :)

I think that the text needs a tiny bit of polishing:

> Define URI query template variables as collection based on DataSchema declarations. The type of the DataSchema cannot be an ObjectSchema or an ArraySchema. 

It is not exactly clear that the keys in the uriVariables map cannot be of ObjectSchema or ArraySchema, i.e. it reads as if uriVariables cannot be an object, it always is. Not sure but maybe something like:

->  Define URI query template variables as a collection based on DataSchema declarations. The individual variables' (keys') DataSchema cannot be an ObjectSchema or an ArraySchema. 

There was also a missing `a` before collection :)

File: index.html [hunk 1]
```diff
@@ -3640,6 +3641,12 @@ <h3><code>forms</code></h3>
 }
 </pre>
 
+<p>
+    <code>uriVariables</code> are mainly for properties and events. When retrofitting an existing system, it may be necessary to use
+    <code>uriVariables</code> for actions. In general, it is recommended to avoid <code>uriVariables</code> as much as possbile when a 
+    new WoT-based system is designed.
```
Comment (relu91) [inline index.html:3647]: Thanks for making this an explicit statement of the specs.

**Review by relu91 — APPROVED** [review]
Nicely done, except for @egekorkan's comment the PR looks good. Just a side note, in #1068 I mentioned the possibility to merge `uriVariables` in the `input` object for actions avoiding ambiguities.  It is fine that it is not addressed in this PR, but I would open a dedicated issue so that we can discuss this possibility. Maybe it could even be deferred for 2.0. What do you think?

File: index.html (hunk 1 above)
Comment (sebastiankb) [reply index.html:3647]: year, that would be a kind of bigger change and should be more addressed in the TD 2.0 version. I would suggest that you create a new issue and I label it with defer 2.0

**Comment by sebastiankb** [comment]
new commit includes the CR from Ege

**Comment by sebastiankb** [comment]
from today's TD call it was decided to merge this PR

**Pull request merged**